src/commands/deploy.ts

```diff
@@ -152,26 +160,90 @@ export const run: Run<DeployArgs> = async (ctx) => {
         return;
       }
 
-      const { publishStarted, progress: step, issues, status } = publishStatus;
-      if (!printedProblems && issues.length > 0) {
+      const { publishStarted, progress: step, issues, status, deletedModelsAndFields } = publishStatus;
+      const hasIssues = issues.length > 0;
+
+      const { deletedModels, deletedModelFields } = deletedModelsAndFields ?? { deletedModels: [], deletedModelFields: [] };
+      const hasDataLoss = deletedModels.length > 0 || deletedModelFields.length > 0;
+
+      if (!printedProblems && (hasIssues || hasDataLoss)) {
         printedProblems = true;
 
         const fatalIssues = issues.filter((issue) => issue.severity === ProblemSeverity.Fatal);
         if (fatalIssues.length > 0) {
           await reportErrorAndExit(ctx, new DeployDisallowedError(publishIssuesToProblems(fatalIssues)));
         }
 
-        println({ ensureEmptyLineAbove: true, content: chalk.bold("Problems found.") });
-        printProblems({ problems: publishIssuesToProblems(issues) });
+        println({ ensureEmptyLineAbove: true, content: sprint`{bold {yellow !}} {bold Problems found}` });
```

Comment:
nit: You can use `.` to combine chalk attributes

```suggestion
        println({ ensureEmptyLineAbove: true, content: sprint`{bold.yellow !} {bold Problems found}` });
```